plugin/core/communication.py

```
@@ -0,0 +1,790 @@
+                with state.acquire_event_handlers_write_lock():
+                    handler = state.event_handlers.get(f"{plugin_id}.{entry_id}")
+                    if handler and hasattr(handler.meta, "enabled"):
+                        object.__setattr__(handler.meta, "enabled", False)
+                
+                self.logger.info(f"Entry '{entry_id}' disabled for plugin {plugin_id}")
+                
+            else:
+                self.logger.warning(f"Unknown ENTRY_UPDATE action: {action}")
+                
+        except Exception as e:
+            self.logger.exception(f"Failed to handle ENTRY_UPDATE: {e}")
```

Ruff (0.15.1)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)